lib/commands/streamscreen.js

```diff
@@ -0,0 +1,408 @@
+    screenRecordCmd += ` --size=${adjustedWidth}x${adjustedHeight}`;
+  }
+  if (bitRate) {
+    screenRecordCmd += ` --bit-rate=${adjustedBitrate}`;
+  }
+  const adbArgs = [
+    ...adb.executable.defaultArgs,
+    'exec-out',
+    // The loop is required, because by default the maximum record duration
+    // for screenrecord is always limited
+    `while true; do ${screenRecordCmd} -; done`,
+  ];
+  const deviceStreaming = spawn(adb.executable.path, adbArgs);
+  deviceStreaming.on('exit', (code, signal) => {
+    log.debug(`Device streaming process exited with code ${code}, signal ${signal}`);
+  });
+
+  let isStarted = false;
+  const deviceStreamingLogger = logger.getLogger(`${SCREENRECORD_BINARY}@${this.opts.udid}`);
+  const errorsListener = (chunk) => {
+    const stderr = chunk.toString();
+    if (_.trim(stderr)) {
+      deviceStreamingLogger.debug(stderr);
+    }
+  };
+  deviceStreaming.stderr.on('data', errorsListener);
+
+  const startupListener = (chunk) => {
+    if (!isStarted) {
+      isStarted = !_.isEmpty(chunk);
+    }
+  };
+  deviceStreaming.stdout.on('data', startupListener);
+
+  try {
+    log.info(`Starting device streaming: ${quote([adb.executable.path, ...adbArgs])}`);
+    await waitForCondition(() => isStarted, {
+      waitMs: STREAMING_STARTUP_TIMEOUT_MS,
+      intervalMs: 300,
+    });
+  } catch (e) {
+    log.errorAndThrow(
+      `Cannot start the screen streaming process. Original error: ${e.message}`);
+  } finally {
+    deviceStreaming.stderr.removeListener('data', errorsListener);
+    deviceStreaming.stdout.removeListener('data', startupListener);
+  }
+  return deviceStreaming;
+}
+
+async function initGstreamerPipeline (deviceStreamingProc, deviceInfo, opts = {}) {
+  const {
+    width,
+    height,
+    quality,
+    tcpPort,
+    considerRotation,
+    logPipelineDetails,
+  } = opts;
+  const adjustedWidth = parseInt(width, 10) || deviceInfo.width;
+  const adjustedHeight = parseInt(height, 10) || deviceInfo.height;
+  const gstreamerPipeline = new SubProcess(GSTREAMER_BINARY, [
+    '-v',
+    'fdsrc', 'fd=0',
+    '!', 'video/x-h264,' +
+      `width=${considerRotation ? Math.max(adjustedWidth, adjustedHeight) : adjustedWidth},` +
+      `height=${considerRotation ? Math.max(adjustedWidth, adjustedHeight) : adjustedHeight},` +
+      `framerate=${deviceInfo.fps}/1,` +
+      'byte-stream=true',
+    '!', 'h264parse',
+    '!', 'queue', 'leaky=downstream',
+    '!', 'avdec_h264',
+    '!', 'queue', 'leaky=downstream',
+    '!', 'jpegenc', `quality=${quality}`,
+    '!', 'multipartmux', `boundary=${BOUNDARY_STRING}`,
+    '!', 'tcpserversink', `host=${TCP_HOST}`, `port=${tcpPort}`,
+  ], {
+    stdio: [deviceStreamingProc.stdout, 'pipe', 'pipe']
+  });
+  gstreamerPipeline.on('exit', (code, signal) => {
+    log.debug(`Pipeline streaming process exited with code ${code}, signal ${signal}`);
+  });
+  const gstreamerLogger = logger.getLogger(`gstreamer@${this.opts.udid}`);
```

Comment:
This might create relatively unreadable lines, since the prefix will be really long, no?

Reply:
the length of udid is about 12 chars. We can make it shorter if needed in the future